server/src/test/java/org/elasticsearch/search/query/QueryPhaseTests.java

```diff
@@ -141,6 +147,7 @@ private void countTestCase(boolean withDeletions) throws Exception {
         countTestCase(matchAllCsq, reader, false);
         countTestCase(tq, reader, withDeletions);
         countTestCase(tCsq, reader, withDeletions);
+        countTestCase(dvfeq, reader, withDeletions);
```

Comment:
This currently fails because the underlying test method compares the output of a search with size 0 to the output of using the count API. The assumption is in both cases we don't need to collect if there are no deletions (otherwise yes). This currently doesn't work for DocValuesFieldExistsQuery (see 
https://github.com/apache/lucene-solr/blob/master/lucene/core/src/java/org/apache/lucene/search/IndexSearcher.java#L339)

Reply:
@jpountz re #37475, this is where I stumbled upon the use of searcher#count https://github.com/elastic/elasticsearch/pull/39570/files#diff-5fac69dcf1790a41ec05793b9c025862R109

Reply:
Maybe this test should be changed so we can also use ot with DocValuesFieldExistsQuery?

Reply:
+1 to fix the test